### PR TITLE
Fix spine memory leak when restarting game and fix memory leaks in js_se_setExceptionCallback and nativevalue_to_se(const cc::Data &from, ...)

### PR DESCRIPTION
**Pull request by dumganhar**
-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Review by PatriceJiang — APPROVED** [review]

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

File: native/cocos/bindings/jswrapper/v8/Object.cpp [hunk 1]
```diff
@@ -51,6 +51,55 @@ uint32_t nativeObjectId = 0;
...
+        if (nativeObj == nullptr) {
+            // Not a JSB binding object
+            return;
+        }
+
+        // Remove mapping
+        auto iter = se::NativePtrToObjectMap::find(nativeObj);
+        if (iter != se::NativePtrToObjectMap::end()) {
+            se::NativePtrToObjectMap::erase(iter);
+        }
```
Comment (PatriceJiang) [inline native/cocos/bindings/jswrapper/v8/Object.cpp:85]: Call erase directly?

**Review by PatriceJiang — APPROVED** [review]

File: native/cocos/bindings/jswrapper/v8/Object.cpp (hunk 1 above)
Comment (dumganhar) [reply native/cocos/bindings/jswrapper/v8/Object.cpp:85]: Yes, could do that, but use `find` and `erase` is more explicit.

**Review by minggo — APPROVED** [review]

**Pull request merged**